pkg/controlsvc/controlsvc.go

```diff
@@ -356,6 +356,12 @@ func (s *Server) RunControlSvc(ctx context.Context, service string, tlscfg *tls.
 					if ctx.Err() != nil {
 						return
 					}
+					select {
+					case <-ctx.Done():
```

Comment:
ctx.Err() is non-nil when context is closed. does that mean we are covered by the lines just above this select block?

Reply:
You're correct, the logic will remain the same. Good spot

After looking at it some more I can see online that typically `ctx.Err()` is used when we want to return information on why the context was cancelled. 
In this case, nothing is returned, so I'm wondering if I should replace the `Err()` with the `Done()`, 
Any thoughts?